charts/patch/templates/patch.json

```diff
@@ -90,7 +90,11 @@
   {{- end }}
   {
     "op": "add",
+  {{- if .Values.proxy.await }}
+    "path": "{{$prefix}}/spec/containers/0",
```

Comment:
I don't think there should be any issue always making it the first container, but I didn't want to assume for this first change. If for some reason any users do depend on their app container coming up first, then when this is disabled they won't need to edit the ordering.

If we do end up making `proxy.await: true` by default though, this is just another part of the templating we can simplify though.

Reply:
After the discussion that happened below, this will remain conditional. That way, when users disable this feature the proxy container is not also set as the first container in the manifest.